app/controllers/application_controller.rb

```diff
@@ -8,10 +8,18 @@ def render_404
     raise ActionController::RoutingError.new('Not Found')
   end
 
+  def require_login
+    if @login_user.nil?
+      redirect_to github_login_path
```

Comment:
It's interesting that you force them to login if they go to an invalid route.  It works well, but makes the user log in without actively choosing to.